File: Napakalaki.java/src/NapakalakiGame/CultistPlayer.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package NapakalakiGame;

/**
 *
 * @author victor
 */
public class CultistPlayer extends Player {
    private static int totalCultistPlayers = 0;
    private Cultist myCultistCard;
    
    public CultistPlayer(Player p, Cultist c){
        super(p);
        myCultistCard = c;
        totalCultistPlayers++;
    }
    
    @Override
    protected int getCombatLevel () {
        return 3 * super.getCombatLevel() / 2 + myCultistCard.getGainedLevels() * totalCultistPlayers;
    }
    
    @Override
    protected int getOponentLevel(Monster m) {
        return m.getCombatLevelAgainstCultistPlayer();
    }
    
    @Override
    protected boolean shouldConvert() {
        return false;
    }
    
    public static int getTotalCultistPlayers () {
        return totalCultistPlayers;
    }
}

File: Napakalaki.ruby/CultistPlayer.rb

require_relative 'Player'
require_relative 'Cultist'

module NapakalakiGame

class CultistPlayer < Player

  @@totalCultistPlayers = 0

  def initialize (p, c)
    super(p)
    @myCultistCard = c
    @@totalCultistPlayers += 1
  end

protected
  def getCombatLevel
      (3 * super() / 2 + @myCultistCard.getGainedLevels() * @@totalCultistPlayers).to_i
  end

  def getOponentLevel(m)
      m.getCombatLevelAgainstCultistPlayer
  end

  def shouldConvert
    false
  end

public
  def getTotalCultistPlayers
      @@totalCultistPlayers
  end

end

end
